### PR TITLE
Cypress v10

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

- Upgrades Cypress from v7 to v10
    - https://www.cypress.io/blog/2022/06/01/cypress-10-release/
- v10 has new conventions:
    - a `cypress.config.js` configuration file:
        https://docs.cypress.io/guides/references/configuration
    - naming test files - `spec.js` -> `cy.js`
    - two distinct types of tests: e2e and component tests
    - for a full run through here's the migration guide:
        https://docs.cypress.io/guides/references/migration-guide
- Fixed some flakey tests

## Why?

- New features, performance and bug fixes
- v10 introduces (beta) component testing 🎉 
    - https://docs.cypress.io/guides/component-testing/mounting-react    

## Other

There's a fairly horrifying clash between Cypress and Jest global types which need to be separated by excluding `cypress.config.js` in `tsconfig.json`.

https://github.com/cypress-io/cypress/issues/22059

### CODE REVIEW

File: dotcom-rendering/cypress/e2e/parallel-1/article.elements.cy.js [hunk 1]
```diff
@@ -96,7 +96,9 @@ describe('Elements', function () {
 		it('should render the click to view overlay revealing the embed when clicked', function () {
 			const getIframeBody = () => {
 				return cy
-					.get('div[data-cy="embed-block"] > div > iframe')
+					.get('div[data-cy="embed-block"] > div > iframe', {
+						timeout: 30000,
```
Comment (mxdvl) [inline dotcom-rendering/cypress/e2e/parallel-1/article.elements.cy.js:100]:
```suggestion
						timeout: 30_000,
```

**Review by mxdvl — APPROVED** [review]
Thanks for taking the time <3

**Review by OllysCoding — APPROVED** [review]
+1 This is awesome!!

**Review by oliverlloyd — APPROVED** [review]
Awesome work

File: dotcom-rendering/cypress/e2e/parallel-1/article.elements.cy.js (hunk 1 above)
Comment (arelra) [reply dotcom-rendering/cypress/e2e/parallel-1/article.elements.cy.js]: Linter says no :)

```
/home/runner/work/dotcom-rendering/dotcom-rendering/dotcom-rendering/cypress/e2e/parallel-1/article.elements.cy.js
  100:18  error  Parsing error: Identifier directly after number
```

**Pull request merged**